recOrder/preproc/pre_processing.py

```diff
@@ -45,3 +45,34 @@ def brenner_gradient(im):
     focus_idx_max = np.where(focus_scores == np.max(focus_scores))[0][0]
 
     return focus_idx_max, focus_idx_min
+
+def get_autocontrast_limits(img, clip = .01):
+
+    #TODO: Figure out how to do this for float images (ret, phase, etc.)
+    # current only works for uint8, uint16, etc.
+
+    data_type = img.dtype
+
+    # Will raise an error if dtype is float, in that case cap the n_bins at 65536
+    try:
+        n_bins = int(np.iinfo(data_type).max)
+    except:
+        n_bins = 65536
```

Comment:
For uint16 if we want to get the exact gray-level at which the contrast limits should be set, then it is necessary to bin in the number of bins intrinsic to the dtype.  We can do something like you suggest for floats, but otherwise I think it is necessary to keep this level of binning